library/std/src/sys/windows/process.rs

```diff
@@ -552,7 +558,7 @@ impl Stdio {
 
 impl From<AnonPipe> for Stdio {
     fn from(pipe: AnonPipe) -> Stdio {
-        Stdio::Handle(pipe.into_handle())
```

Comment:
Is this line the key bug? `Stdio::Handle` expects a sync pipe but `pipe.into_handle()` produces an async pipe?
That seems like maybe there should be comments in some places so that this is not reverted in the future when the context of this PR has been lost. (But let's add comments, if any, in a separate PR, so that this one can land ASAP.)

Reply:
Good point. I'll add that in a follow up PR.